vsintegration/src/FSharp.Editor/Formatting/IndentationService.fs

```diff
@@ -40,7 +41,22 @@ type internal FSharpIndentationService() =
                         | ' ' -> loop (column + 1) (spaces + 1)
                         | '\t' -> loop (column + 1) (((spaces / tabSize) + 1) * tabSize)
                         | _ -> spaces
-                Some (loop 0 0)
+
+                let lastIndent = loop 0 0
+
+                // Increase indent after line end with:
+                let regex =
+                    @"(
+                        = | <-
+                       | ->
+                       | \( | \[ | \[\| | \[< | \{
+                       | begin | do | class | function | then | else | struct | try
+                      )
+                      \s*$"
+                if Regex.IsMatch(previousLine.ToString(), regex, RegexOptions.IgnorePatternWhitespace) then
```

Comment:
@KevinRansom 
A better approach should be like @saul said, using tokenizer but I honestly don't know how.
I think comments end with the above keywords are not as common as normal code though.